crates/paimon/src/io/file_io.rs

```diff
@@ -0,0 +1,186 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use crate::error::*;
+use std::collections::HashMap;
+
+use opendal::services::MemoryConfig;
+use opendal::{Metakey, Operator};
+use snafu::ResultExt;
+
+#[derive(Clone, Debug)]
+pub struct FileIO {
+    op: Operator,
+}
+
+impl FileIO {
+    /// Create a new FileIO.
+    ///
+    /// The input HashMap is paimon-java's [`Options`](https://github.com/apache/paimon/blob/405779a83e36bfb27af4870a3e710ddbdff729a4/paimon-common/src/main/java/org/apache/paimon/options/Options.java#L60)
```

Comment:
Could the link use the source code of certain version to help to follow in feature?

Reply:
Fixed to 0.8.2